types/src/transaction/program.rs

```diff
@@ -162,3 +185,62 @@ impl fmt::Debug for TransactionArgument {
         }
     }
 }
+
+impl CanonicalSerialize for TransactionArgument {
+    fn serialize(&self, serializer: &mut impl CanonicalSerializer) -> Result<()> {
+        match self {
+            TransactionArgument::U64(value) => {
+                serializer.encode_u32(TransactionArgument_ArgType::U64 as u32)?;
+                serializer.encode_u64(*value)?;
+            }
+            TransactionArgument::Address(address) => {
+                serializer.encode_u32(TransactionArgument_ArgType::ADDRESS as u32)?;
+                serializer.encode_struct(address)?;
+            }
+            TransactionArgument::String(string) => {
+                serializer.encode_u32(TransactionArgument_ArgType::STRING as u32)?;
+                serializer.encode_vec(string.as_bytes())?;
```

Comment:
This can be updated to encode_string() now?

Reply:
Yes!